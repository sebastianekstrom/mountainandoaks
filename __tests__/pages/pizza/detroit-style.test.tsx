import { render, screen } from "@testing-library/react";
import React from "react";

import DetroitStyle from "pages/pizza/detroit-style";

describe("DetroitStyle page", () => {
  it("renders title correctly", () => {
    render(<DetroitStyle />);
    const title = screen.queryAllByText(/Detroit style pizza/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<DetroitStyle />);
    const ingress = screen.queryAllByText(
      /The Detroit style pizza is a deep-dish pizza/i
    );
    expect(ingress).toBeDefined();
  });

  it("renders the correct ingredients", () => {
    render(<DetroitStyle />);
    const ingredientList = screen.getByTestId("INGREDIENT_LIST");
    expect(ingredientList).toMatchSnapshot();
  });

  it("renders the method steps", () => {
    render(<DetroitStyle />);
    const step = screen.queryAllByText(
      /Adjust oven rack to lowest position and preheat/i
    );
    expect(step).toBeDefined();
  });

  it("renders the alert with the correct description", () => {
    render(<DetroitStyle />);
    const alert = screen.queryAllByText(/Remember to over season your sauce/i);
    expect(alert).toBeDefined();
  });
});
