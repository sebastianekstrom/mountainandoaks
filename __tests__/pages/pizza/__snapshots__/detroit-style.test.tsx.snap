// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetroitStyle page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="mt-8"
  >
    <div
      class="mb-2"
    >
      <h3
        class="variable font-heading text-h3 leading-h3 text-regular "
      >
        Dough
      </h3>
    </div>
    <div
      class="flex flex-col"
    >
      <ul>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Bread flour
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            300
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Water
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            225
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Salt
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            11
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Dry yeast
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            5
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Olive oil to coat the pan
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            30
            g
          </p>
        </li>
      </ul>
      <div
        class="ml-auto"
      >
        <button
          class="mt-2 rounded-full text-regular bg-pill-background py-2 px-4"
          data-testid="BAKERS_PERCENTAGE"
          role="button"
          type="button"
        >
          <p
            class="variable font-sans text-caption leading-paragraph "
          >
            Show bakers %
          </p>
        </button>
      </div>
    </div>
  </div>
  <div
    class="mt-8"
  >
    <div
      class="mb-2"
    >
      <h3
        class="variable font-heading text-h3 leading-h3 text-regular "
      >
        Tomato sauce
      </h3>
    </div>
    <div
      class="flex flex-col"
    >
      <ul>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Canned tomatoes
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            400
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Olive oil
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            30
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Salt
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            7
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Sugar
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            7
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Dried oregano
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            3
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Garlic powder
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            3
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Onion powder
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            3
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Cloves of garlic
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            1.5
          </p>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="mt-8"
  >
    <div
      class="mb-2"
    >
      <h3
        class="variable font-heading text-h3 leading-h3 text-regular "
      >
        Toppings
      </h3>
    </div>
    <div
      class="flex flex-col"
    >
      <ul>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Mozzarella (low moisture)
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            170
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Cheeze of your choice
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            170
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Sausage (pepperoni or chorizo)
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            340
            g
          </p>
        </li>
        <li
          class="flex justify-between mb-1 transition-opacity "
        >
          <div
            class=""
          >
            <p
              class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
            >
              Finely shredded Parmesan
            </p>
          </div>
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph "
          >
            20
            g
          </p>
        </li>
      </ul>
    </div>
  </div>
</section>
`;
